core/src/main/scala/org/bitcoins/core/psbt/PSBT.scala

```diff
@@ -494,6 +512,30 @@ case class PSBT(
     PSBT(globalMap, newInputMaps, outputMaps)
   }
 
+  def addSignatures(
+      partialSignatures: Vector[PartialSignature],
+      inputIndex: Int): PSBT = {
+    require(
+      inputIndex < inputMaps.size,
+      s"index must be less than the number of input maps present in the psbt, $inputIndex >= ${inputMaps.size}")
+    require(
+      !inputMaps(inputIndex).isFinalized,
+      s"Cannot update an InputPSBTMap that is finalized, index: $inputIndex")
+    val intersect = inputMaps(inputIndex).partialSignatures
+      .map(_.pubKey)
+      .intersect(partialSignatures.map(_.pubKey))
+    require(
```

Comment:
do we really want to fail in this case? It seems like maybe we should just ignore?

Reply:
Changed to only throw error if the signatures are different